doc/wg/core/notes/core-notes-2020-12-11.md

```diff
@@ -0,0 +1,95 @@
+# Tock Core Notes 2020-12-11
+
+## Attending
+ * Branden Ghena
+ * Amit Levy
+ * Leon Schuermann
+ * Arjun Deopujari
+ * Johnathan Van Why
+ * Vadim Sukhomlinov
+ * Pat Pannuto
+ * Brad Campbell
+ * Phil Levis
+ * Hudson Ayers
+ * Alistair
+
+
+## Updates
+- Phil: Progress on porting
+
+- Arjun: Working on DALS over-the-air-updates implementation.
+
+- Brad: Worked on Rubble, cleaning up the stack.
+- Amit: Advertisement issue on Android?
+- Brad: Still outstanding.
+- Hudson: Issue with scan response.
+- Brad: Not clear if Rubble API will let us call the function we need.
+
+## System Call Porting
+- Amit: Sign up here: https://github.com/tock/tock/issues/2235
+- Phil: I grouped them, useful to get experience with porting them and learning
+  new interface.
+- Amit: Some are harder to test without hardware.
+- Amit: I can take on the BLE advertisements capsule.
+- Leon: We are just doing direct translation now, not trying to take advantage
+  of the new capabilities.
+
+## App Slice Swapping
+- Leon: Should the kernel guarantee that capsules behave correctly with app
+  slice management?
+- Johnathan: Kernel must ensure capsules cannot duplicate app slices and pass
+  them back multiple times. Not as much of an issue if the capsule switches them
+  around.
+- Johnathan: Should the kernel give capsules a method for accessing app slices
+  they have access to rather than a reference to the app slice?
+- Leon: The kernel could create app slices as needed, could lead to linear
+  searches or a fair bit of complexity.
+- Phil: End-to-end argument would say userspace has to check anyway, so is there
+  a threat model we have to protect against?
+- Johnathan: If the kernel provides guarantee, we do not need to do libtock-rs
+  check in userspace.
+- Leon: Could benefit application correctness.
+- Johnathan: Might have to check for null anyway, not much overhead to check for
+  buffer correctness as well.
+
+- Phil: Idea: Capsule maintains table of appslices, and uses allow number to
+  index the table and therefore has to swap the buffer (since that is all you
+  could access).
+- Leon: Tried this, couldn't get it to work in Tock context.
```

Comment:
What I wanted to say here was that this approach is doable, but has a significant overhead (as you'd have to traverse through a range of pointers prior to knowing the appslice address and length), instead of containing this information in a high-level Rust type that you own.

Sorry, might have mixed some things up in the discussion. The statement is still correct, so we can leave this in. :smile: